tests/slsa_analyzer/package_registry/test_osv_dev.py

```diff
@@ -0,0 +1,220 @@
+# Copyright (c) 2025 - 2025, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""Tests for the osv.dev service."""
+
+import os
+from pathlib import Path
+
+import pytest
+
+from macaron.config.defaults import load_defaults
+from macaron.errors import APIAccessError
+from macaron.slsa_analyzer.package_registry.osv_dev import OSVDevService
+
+
+@pytest.mark.parametrize(
+    ("user_config_input"),
+    [
+        pytest.param(
+            """
+            [osv_dev]
+            url_netloc =
+            url_scheme = https
+            query_endpoint = v1/query
+            """,
+            id="Missing netloc",
+        ),
+        pytest.param(
+            """
+            [osv_dev]
+            url_netloc = osv.dev
+            url_scheme = https
+            query_endpoint =
+            """,
+            id="Missing query endpoint",
+        ),
+    ],
+)
+def test_load_defaults_query_api(tmp_path: Path, user_config_input: str) -> None:
+    """Test the ``load_defaults`` method."""
+    user_config_path = os.path.join(tmp_path, "config.ini")
+
+    with open(user_config_path, "w", encoding="utf-8") as user_config_file:
+        user_config_file.write(user_config_input)
+
+    # We don't have to worry about modifying the ``defaults`` object causing test
+    # pollution here, since we reload the ``defaults`` object before every test with the
+    # ``setup_test`` fixture.
+    load_defaults(user_config_path)
+
+    with pytest.raises(APIAccessError):
+        OSVDevService.call_osv_query_api({})
+
+
+def test_is_affected_version_invalid_commit() -> None:
+    """Test if the function can handle invalid commits"""
+    with pytest.raises(APIAccessError):
+        OSVDevService.is_version_affected(
+            vuln={}, pkg_name="pkg", pkg_version="invalid_commit", ecosystem="GitHub Actions"
+        )
+
+
+def test_is_affected_version_invalid_response() -> None:
+    """Test if the function can handle empty OSV response."""
+    with pytest.raises(APIAccessError):
+        OSVDevService.is_version_affected(
+            vuln={"vulns": []}, pkg_name="repo/workflow", pkg_version="1.0.0", ecosystem="GitHub Actions"
+        )
```

Comment:
To me it seems that both of these functions fail because `vuln` is not correctly populated. In `is_version_affected` we have the following block:
```
affected = json_extract(vuln, ["affected"], list)
if not affected:
    raise APIAccessError(f"Failed to extracted info for {pkg_name}@{pkg_version}.")
```
Can we be sure that the error is being raised from the expected part of the function?

Reply:
See commit [a684508](https://github.com/oracle/macaron/pull/1021/commits/a6845083b70a170af22416b1f8cce556e16ca5dc).